File: lib/providers/externals/userapi/endpoints.dart
import 'package:fuelify/providers/externals/userapi/model.dart';

class UserApiEndpoints {
  static const String _baseURL = 'http://10.0.2.2:3000'; //'http://localhost:3000'; 
  static const String _apiVersion = '/api/v1';

  static ApiEndpoint get login => ApiEndpoint(
    url: '$_baseURL$_apiVersion/user/login',
    method: 'POST',
    description: '[POST] User request to login with provided credentials',
    requiresAuthentication: false,
  );

  static ApiEndpoint get logout => ApiEndpoint(
    url: '$_baseURL$_apiVersion/user/logout',
    method: 'POST',
    description: '[POST] User request to logout',
    requiresAuthentication: true,
  );

  static ApiEndpoint get getUser => ApiEndpoint(
    url: '$_baseURL$_apiVersion/user',
    method: 'GET',
    description: '[GET] User account data',
    requiresAuthentication: true,
  );

  static ApiEndpoint get getMealPlan => ApiEndpoint(
    url: '$_baseURL$_apiVersion/plan',
    method: 'GET',
    description: '[GET] User meal plan for a date range',
    //requiresAuthentication: true,
  );

  static ApiEndpoint get getDayMealPlan => ApiEndpoint(
    url: '$_baseURL$_apiVersion/plan/day',
    method: 'GET',
    description: '[GET] User meal plan for a specific date',
    requiresAuthentication: true,
  );

  static ApiEndpoint get deleteMeal => ApiEndpoint(
    url: '$_baseURL$_apiVersion/plan/meal',
    method: 'DELETE',
    description: '[DELETE] Meal from plan',
    requiresAuthentication: true,
  );

  static ApiEndpoint get addMeal => ApiEndpoint(
    url: '$_baseURL$_apiVersion/plan/meal',
    method: 'PUT',
    description: '[PUT] Add meal to plan',
    requiresAuthentication: true,
  );

  static ApiEndpoint get getMeal => ApiEndpoint(
    url: '$_baseURL$_apiVersion/plan/meal',
    method: 'GET',
    description: '[GET] Get plan meal details',
    requiresAuthentication: true,
  );

  static ApiEndpoint get updateMeal => ApiEndpoint(
    url: '$_baseURL$_apiVersion/plan/meal/update',
    method: 'POST',
    description: '[POST] Update meal in calendar',
    requiresAuthentication: true,
  );

  static ApiEndpoint get moveMeal => ApiEndpoint(
    url: '$_baseURL$_apiVersion/plan/meal/move',
    method: 'POST',
    description: '[POST] Move meal in calendar',
    requiresAuthentication: true,
  );

  static ApiEndpoint get addMealFeedback => ApiEndpoint(
    url: '$_baseURL$_apiVersion/plan/meal/feedback',
    method: 'PUT',
    description: '[PUT] Add feedback to meal',
    requiresAuthentication: true,
  );

  static ApiEndpoint get deleteMealFeedback => ApiEndpoint(
    url: '$_baseURL$_apiVersion/plan/meal/feedback',
    method: 'DELETE',
    description: '[DELETE] Delete feedback from meal',
    requiresAuthentication: true,
  );

  static ApiEndpoint get getRecipe => ApiEndpoint(
    url: '$_baseURL$_apiVersion/recipe',
    method: 'GET',
    description: '[GET] Recipe details',
    requiresAuthentication: true,
  );

  static ApiEndpoint get addRecipe => ApiEndpoint(
    url: '$_baseURL$_apiVersion/recipe',
    method: 'PUT',
    description: '[PUT] Add a new recipe',
    requiresAuthentication: true,
  );

  static ApiEndpoint get classifyRecipe => ApiEndpoint(
    url: '$_baseURL$_apiVersion/recipe/classification',
    method: 'PUT',
    description: '[PUT] Recipe classification (discovery - like, dislike, favorite)',
    requiresAuthentication: true,
  );

  static ApiEndpoint get addRecipeToFavorites => ApiEndpoint(
    url: '$_baseURL$_apiVersion/recipe/classification/update',
    method: 'POST',
    description: '[POST] Recipe classification update',
    requiresAuthentication: true,
  );

  static ApiEndpoint get getClassifiedRecipes => ApiEndpoint(
    url: '$_baseURL$_apiVersion/recipe/classification',
    method: 'GET',
    description: '[GET] Get classified recipes (provide query parameter to get individually - liked|disliked|favorited or collectively)',
    requiresAuthentication: true,
  );

  static ApiEndpoint get deleteRecipe => ApiEndpoint(
    url: '$_baseURL$_apiVersion/recipe',
    method: 'DELETE',
    description: '[DELETE] Delete a recipe (only deletes from personal collection not from database)',
    requiresAuthentication: true,
  );

  static ApiEndpoint get getReceipes => ApiEndpoint(
    url: '$_baseURL$_apiVersion/recipe/search',
    method: 'GET',
    description: '[GET] Search recipes according to filtering criteria',
    requiresAuthentication: true,
  );

  static ApiEndpoint get getSimilarRecipes => ApiEndpoint(
    url: '$_baseURL$_apiVersion/recipe/similar',
    method: 'GET',
    description: '[GET] Similar recipes to provided recipe',
    requiresAuthentication: true,
  );

  static ApiEndpoint get addRecipeReview => ApiEndpoint(
    url: '$_baseURL$_apiVersion/recipe/review',
    method: 'PUT',
    description: '[PUT] Add review on recipe',
    requiresAuthentication: true,
  );
}

File: react-app/packages/shared/src/api/endpoints.ts
// API endpoints — DynamoDB-backed (auth & user settings only)
// Meal plans, recipes, and food data now go through Supabase directly.

export interface ApiEndpoint {
  url: string;
  method: 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH';
  description: string;
  requiresAuthentication: boolean;
}

// Base URL configuration — mirrors AppUrl from Flutter
const LIVE_BASE_URL = 'http://18.117.207.252';
const LOCAL_BASE_URL = 'http://localhost:3000';

// Toggle this for dev vs prod
export const BASE_URL = LOCAL_BASE_URL;
export const API_VERSION = '/api/v1';
const API_BASE = `${BASE_URL}${API_VERSION}`;

// Legacy endpoints (from dependencies/endpoints.dart)
export const LegacyEndpoints = {
  login: `${BASE_URL}/user/login`,
  register: `${BASE_URL}/user/register`,
  forgotPassword: `${BASE_URL}/forgot-password`,
  testToken: `${BASE_URL}/tokentest`,
  userUpdateProfile: `${BASE_URL}/user/profile/update`,
  userUpdateOnboardingState: `${BASE_URL}/user/state/onboarding`,
} as const;

// ===================================================================
// DynamoDB-backed endpoints (auth, user account, settings, onboarding)
// These continue to go through the existing Node/Express API → DynamoDB
// ===================================================================
export const Endpoints = {
  // Auth
  login: {
    url: `${API_BASE}/user/login`,
    method: 'POST',
    description: '[POST] User request to login with provided credentials',
    requiresAuthentication: false,
  },
  logout: {
    url: `${API_BASE}/user/logout`,
    method: 'POST',
    description: '[POST] User request to logout',
    requiresAuthentication: true,
  },

  // User account & settings (DynamoDB)
  getUser: {
    url: `${API_BASE}/user`,
    method: 'GET',
    description: '[GET] User account data',
    requiresAuthentication: true,
  },
  updateProfile: {
    url: `${BASE_URL}/user/profile/update`,
    method: 'POST',
    description: '[POST] Update user profile (DynamoDB)',
    requiresAuthentication: true,
  },
  updateOnboardingState: {
    url: `${BASE_URL}/user/state/onboarding`,
    method: 'POST',
    description: '[POST] Update user onboarding state (DynamoDB)',
    requiresAuthentication: true,
  },
} as const satisfies Record<string, ApiEndpoint>;

// ===================================================================
// Supabase-backed data (meal plans, recipes, food, reviews)
// These no longer go through the API — they use the Supabase client
// directly via MealPlanRepository and RecipeRepository.
//
// Migrated endpoints (for reference):
//   GET    /api/v1/plan              → MealPlanRepository.getMealPlan()
//   GET    /api/v1/plan/day          → MealPlanRepository.getDayMealPlan()
//   PUT    /api/v1/plan/meal         → MealPlanRepository.addMeal()
//   GET    /api/v1/plan/meal         → (included in getMealPlan results)
//   POST   /api/v1/plan/meal/update  → MealPlanRepository.updateMeal()
//   POST   /api/v1/plan/meal/move    → MealPlanRepository.moveMeal()
//   DELETE /api/v1/plan/meal         → MealPlanRepository.deleteMeal()
//   PUT    /api/v1/plan/meal/feedback   → MealPlanRepository.addMealFeedback()
//   DELETE /api/v1/plan/meal/feedback   → MealPlanRepository.deleteMealFeedback()
//   GET    /api/v1/recipe            → RecipeRepository.getRecipe()
//   PUT    /api/v1/recipe            → RecipeRepository.addRecipe()
//   DELETE /api/v1/recipe            → RecipeRepository.deleteRecipe()
//   GET    /api/v1/recipe/search     → RecipeRepository.searchRecipes()
//   GET    /api/v1/recipe/similar    → RecipeRepository.getSimilarRecipes()
//   PUT    /api/v1/recipe/classification      → RecipeRepository.classifyRecipe()
//   POST   /api/v1/recipe/classification/update → RecipeRepository.updateClassification()
//   GET    /api/v1/recipe/classification       → RecipeRepository.getClassifiedRecipes()
//   PUT    /api/v1/recipe/review     → RecipeRepository.addReview()
// ===================================================================
